doc/asciidoc/docker/data/docker-compose-source-sink.yml

```diff
@@ -2,7 +2,7 @@ version: '3'
 
 services:
   neo4j-source:
-    image: neo4j:3.5
+    image: neo4j:4.0
```

Comment:
throughout in this PR, may as well use the latest docker image that's available.  I believe that's 4.0.2, which fixed some important bugs.

(This comment applies to the earlier docker-compose file too)